src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -661,130 +530,107 @@ private static void cloneImpl(String gitUrl, String scmTag, File checkoutDirecto
-        File parentFile = pluginPath.getParentFile();
-        if (parentFile == null) {
-            return null;
+        String name = gitUrl.substring(++index);
+        if (name.endsWith(".git")) {
+            return name.substring(0, name.length() - 4);
         }
-        File log = new File(parentFile.getAbsolutePath() + File.separatorChar + "modules.log");
-        runner.run(
-                Map.of("expression", "project.modules", "output", log.getAbsolutePath()),
-                parentFile,
-                null,
-                "-q",
-                "help:evaluate");
-        List<String> lines;
-        try {
-            lines = Files.readAllLines(log.toPath(), Charset.defaultCharset());
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        return name;
+    }
+
+    static class RunAndMapBeforeCheckoutHooks implements Function<PluginMetadata, PluginMetadata> {
+
+        private PluginCompatTesterHooks pcth;
+        private String coreVersion;
+        private PluginCompatTesterConfig config;
+
+        RunAndMapBeforeCheckoutHooks(
+                PluginCompatTesterHooks pcth, String coreVersion, PluginCompatTesterConfig config) {
+            this.pcth = pcth;
+            this.coreVersion = coreVersion;
+            this.config = config;
         }
-        for (String line : lines) {
-            if (!StringUtils.startsWith(line.trim(), "<string>")) {
-                continue;
-            }
-            String mvnModule =
-                    line.replace("<string>", "").replace("</string>", "").trim();
-            if (mvnModule != null && mvnModule.contains(module)) {
-                return mvnModule;
+
+        @Override
+        public PluginMetadata apply(PluginMetadata pluginMetadata) throws WrappedPluginCompatibilityException {
+            BeforeCheckoutContext c = new BeforeCheckoutContext(pluginMetadata, coreVersion, config);
+            try {
+                pcth.runBeforeCheckout(c);
+            } catch (PluginCompatibilityTesterException e) {
+                throw new WrappedPluginCompatibilityException(e);
             }
+            return c.getPluginMetadata();
+        }
+    }
+
+    /**
+     * Throws {@code current} if {@code throwException} is {@code true} or returns {@cod caught}
+     * with {@code current} added (if non-null) as a supressed exception.
+     *
+     * @param <T>
+     * @param current the PluginCompatibilityTesterException if any
+     * @param caught the newly caught exception
+     * @param throwException {@code true} if we should immediatly rethrow {@code caught}, {@code
+     *     false} indicating we should return {@caught}.
+     * @return {@code caught}
+     * @throws PluginCompatibilityTesterException if {@code ethrowException == true} then {@caught}
+     *     is thrown.
+     */
+    private static <T extends PluginCompatibilityTesterException> T throwOrAddSupressed(
```

Comment:
This is a nice refactoring but is unnecessarily cluttering up this PR. Can we file a PR against the main branch to implement this cleanup there so that this diff shrinks and is easier to understand for readers?

Reply:
sure.